tests/git.zunit

```diff
@@ -1355,6 +1355,632 @@
+        assert ${lines[3]} same_as "${fg[yellow]}--shallow-exclude=v1             $reset_color 1st commit"
+        assert ${lines[4]} same_as "${fg[yellow]}--shallow-exclude=v2             $reset_color  2nd commit"
+        assert ${lines[5]} same_as "${fg[yellow]}--shallow-exclude=3e209a3        $reset_color 1st commit"
+    }
+
+    prefix=--shallow-exclude=
+    _fzf_complete_git 'git pull '
+}
+
+@test 'Testing completion: git pull --shallow-exclude **' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull --shallow-exclude '
+
+        run cat
+        assert ${#lines} equals 5
+        assert ${lines[1]} same_as "${fg[yellow]}another-branch $reset_color  2nd commit"
+        assert ${lines[2]} same_as "${fg[yellow]}master         $reset_color 1st commit"
+        assert ${lines[3]} same_as "${fg[yellow]}v1             $reset_color 1st commit"
+        assert ${lines[4]} same_as "${fg[yellow]}v2             $reset_color  2nd commit"
+        assert ${lines[5]} same_as "${fg[yellow]}3e209a3        $reset_color 1st commit"
+    }
+
+    prefix=
+    _fzf_complete_git 'git pull --shallow-exclude '
+}
+
+@test 'Testing completion: git pull --negotiation-tip=**' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull '
+
+        run cat
+        assert ${#lines} equals 5
+        assert ${lines[1]} same_as "${fg[yellow]}--negotiation-tip=another-branch $reset_color  2nd commit"
+        assert ${lines[2]} same_as "${fg[yellow]}--negotiation-tip=master         $reset_color 1st commit"
+        assert ${lines[3]} same_as "${fg[yellow]}--negotiation-tip=v1             $reset_color 1st commit"
+        assert ${lines[4]} same_as "${fg[yellow]}--negotiation-tip=v2             $reset_color  2nd commit"
+        assert ${lines[5]} same_as "${fg[yellow]}--negotiation-tip=3e209a3        $reset_color 1st commit"
+    }
+
+    prefix=--negotiation-tip=
+    _fzf_complete_git 'git pull '
+}
+
+@test 'Testing completion: git pull --negotiation-tip **' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull --negotiation-tip '
+
+        run cat
+        assert ${#lines} equals 5
+        assert ${lines[1]} same_as "${fg[yellow]}another-branch $reset_color  2nd commit"
+        assert ${lines[2]} same_as "${fg[yellow]}master         $reset_color 1st commit"
+        assert ${lines[3]} same_as "${fg[yellow]}v1             $reset_color 1st commit"
+        assert ${lines[4]} same_as "${fg[yellow]}v2             $reset_color  2nd commit"
+        assert ${lines[5]} same_as "${fg[yellow]}3e209a3        $reset_color 1st commit"
+    }
+
+    prefix=
+    _fzf_complete_git 'git pull --negotiation-tip '
+}
+
+@test 'Testing completion: git pull -o **' {
```

Comment:
Please add a test that examines the completion in `git pull -qo **`. This will test if it completes in the expected way if the short option is stacked like `-qo`.